File: PipelineService/Models/MqttMessages/OperationExecutionMessage.cs
using System;
using System.Collections.Generic;
using PipelineService.Models.Pipeline;

namespace PipelineService.Models.MqttMessages
{
	public class OperationExecutionMessage : BaseMqttMessage
	{
		/// <summary>
		/// The pipeline's id the node was executed for.
		/// </summary>
		public Guid PipelineId { get; set; }

		/// <summary>
		/// The pipeline execution this node belongs to.
		/// </summary>
		public Guid ExecutionId { get; set; }

		/// <summary>
		/// The id of an operation (node) within a pipeline and execution.
		/// </summary>
		public Guid OperationId { get; set; }

		/// <summary>
		/// The operation's id that will be executed.
		/// </summary>
		public Guid WorkerOperationId { get; set; }

		/// <summary>
		/// An identifier of the operation.
		/// Used by generic pandas operation to execute correct function.
		/// </summary>
		public string WorkerOperationIdentifier { get; set; }

		/// <summary>
		/// The dataset used as inputs for this operation.
		/// </summary>
		public IList<Dataset> Inputs { get; set; }

		/// <summary>
		/// The datasets that will be produced by this operation.
		/// </summary>
		public IList<Dataset> Outputs { get; set; }

		/// <summary>
		/// The key the resulting dataset will be stored as.
		/// </summary>
		[Obsolete("Use Output")]
		public string ResultKey { get; set; }

		/// <summary>
		/// The configuration of the operation.
		/// </summary>
		public IDictionary<string, string> OperationConfiguration { get; set; } = new Dictionary<string, string>();
	}
}

File: OperationWorker/src/models/operation_execution_message.py
from src.models.dataset import Dataset
from src.models.message import Message


class OperationExecutionMessage(Message):

	def __init__(self, deserialized: dict):
		super().__init__()
		self._pipeline_id: str = deserialized['PipelineId']
		self._execution_id: str = deserialized['ExecutionId']
		self._operation_id = deserialized['OperationId']
		self._worker_operation_id = deserialized['WorkerOperationId']
		self._worker_operation_identifier: str = deserialized['WorkerOperationIdentifier']
		self._operation_configuration: dict = deserialized['OperationConfiguration']
		self._inputs: [Dataset] = Dataset.from_array(deserialized['Inputs'])
		self._outputs: [Dataset] = Dataset.from_array(deserialized['Outputs'])

	def get_pipeline_id(self) -> str:
		return self._pipeline_id

	def get_operation_id(self) -> str:
		return self._operation_id

	def get_worker_operation_id(self) -> str:
		return self._worker_operation_id

	def get_worker_operation_identifier(self):
		return self._worker_operation_identifier

	def get_execution_id(self) -> str:
		return self._execution_id

	def get_operation_configuration(self) -> dict:
		return self._operation_configuration

	def get_inputs(self) -> [Dataset]:
		return self._inputs

	def get_outputs(self) -> [Dataset]:
		return self._outputs
